### PR TITLE
Add dev server, new rules, new README, etc.

### PR DESCRIPTION
**Pull request by matanlurey**
Originally https://github.com/dart-lang/bazel/pull/22.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,23 +1,115 @@
...
+```gitignore
+/bazel-*
+.bazelify
+packages.bzl
+BUILD
+WORKSPACE
+```
+
+You may also want to exclude the `bazel-*` folders from the Dart analyzer
+using an `analysis_options.yaml` file. This prevents the Dart analyzer from
```
Comment (jakemac53) [inline README.md]: should this be `.analysis_options`?

Comment (matanlurey) [inline README.md]: Sorry, I am used to the internal version which doesn't use hidden files.

File: README.md [hunk 2]
```diff
@@ -1,23 +1,115 @@
 # bazel
```
Comment (kevmoo) [inline README.md]: Remove this top line. Redundant is all contexts

File: README.md [hunk 3]
```diff
@@ -1,23 +1,115 @@
...
+```
+my_new_package/
+  bin/
+  lib/
+  web/
+  pubspec.yaml
+```
+
+```bash
+pub global run bazel:bazelify -p my_new_package
```
Comment (kevmoo) [inline README.md]: is `-p my_new_package` necessary if run from within a package dir?

Perhaps

```bash
my_pkg_dir/ $ pub global run bazelify
```

Note, you don't need `bazel:` for globally activated binaries

File: lib/src/bazelify/build.dart [hunk 4]
```diff
@@ -282,14 +289,33 @@ class DartWebApplication implements DartBuildRule {
...
     } else {
-      return buffer +
-          ' + [\n' +
+      buffer += ' + [\n' +
           includeLibraries
               .map/*<String>*/((l) => '        ":${l.name}",\n')
               .join() +
           '    ],\n)';
     }
+    buffer += '\ndev_server(\n'
```
Comment (kevmoo) [inline lib/src/bazelify/build.dart]: For future, we can use triple-quote blocks to include newlines. Is cleaner

File: pubspec.yaml [hunk 5]
```diff
@@ -1,5 +1,6 @@
 name: bazel
 description: Bazel support for Dart
+version: 1.0.0-alpha
```
Comment (kevmoo) [inline pubspec.yaml]: Add homepage & author, please

**Review by kevmoo — CHANGES_REQUESTED** [review]

File: lib/src/bazelify/build.dart [hunk 6]
```diff
@@ -282,14 +289,33 @@ class DartWebApplication implements DartBuildRule {
...
+    buffer += '\ndart_ddc_bundle(\n'
+        '    name = "${name}_ddc_bundle",\n'
+        '    entry_library = "$scriptFile",\n'
+        '    entry_module = ":$package",\n'
+        '    input_html = "web/index.html",\n'
+        '    output_dir = "web",\n'
+        ')';
+    buffer += '\ndev_server(\n'
+        '    name = "${name}_ddc_serve",\n'
+        '    deps = [":${name}_ddc_bundle"],\n'
```
Comment (jakemac53) [inline lib/src/bazelify/build.dart]: All you should need here is a `name` and `data = [":${name}_ddc_bundle"]`. The bundle rule includes in its runfiles all the transitive dart sources, sourcemaps, etc (this was fixed recently).

File: README.md (hunk 2 above)
Comment (matanlurey) [inline README.md]: Done

File: README.md (hunk 3 above)
Comment (matanlurey) [inline README.md]: Got it. Made it more clear by `cd`ing in and just running then.

File: test/goldens/build_file_web_application [hunk 7]
```diff
@@ -28,3 +31,22 @@ dart_web_application(
...
+dev_server(
+    name = "main_web_dartium_serve",
+    deps = _PUB_DEPS + [
+        ":silly_monkey",
+    ],
+    data = glob(["web/**"]),
+)
+dart_ddc_bundle(
+    name = "main_web_ddc_bundle",
+    entry_library = "web/main.dart",
```
Comment (jakemac53) [inline test/goldens/build_file_web_application]: This isn't a part of `:silly_monkey`. I think we need to materialize an additional dart_library which has as its sources `glob(["web/**"])`? Setting up the correct dependencies would also require reading web/main.dart and examining its imports or something along those lines....

File: lib/src/bazelify/build.dart (hunk 4 above)
Comment (matanlurey) [inline lib/src/bazelify/build.dart]: Future I'd rather just make this an actual class with a string buffer, but got lazy :)

File: pubspec.yaml (hunk 5 above)
Comment (matanlurey) [inline pubspec.yaml]: Done.

File: test/goldens/build_file_web_application (hunk 7 above)
Comment (jakemac53) [inline test/goldens/build_file_web_application]: or actually for the deps you could just be safe and include all the same deps as `:silly_monkey` plus `:silly_monkey`...

**Comment by kevmoo** [comment]
@matanlurey you're going to squash, right? 😄

File: test/goldens/build_file_web_application (hunk 7 above)
Comment (matanlurey) [inline test/goldens/build_file_web_application]: Ugh. OK, I can hack it for now by emitting:

```build
dart_library(
  name = ...,
  srcs = glob(["lib/**"]) + glob(["web/**"]),
)
```

Long-term we should just have `dart_web_application` handle this.

Comment (jakemac53) [inline test/goldens/build_file_web_application]: agreed

**Comment by matanlurey** [comment]
I'll squash + merge, don't worry :)

**Pull request merged**